src/coreclr/vm/interpexec.cpp

```diff
@@ -1252,6 +1252,31 @@ void InterpExecMethod(InterpreterFrame *pInterpreterFrame, InterpMethodContextFr
 
                     goto CALL_INTERP_SLOT;
                 }
+                case INTOP_NEWOBJ_ARR:
+                {
+                    const int32_t dVar = ip[1];
+                    const int32_t callArgsOffset = ip[2];
+                    const int32_t clsIdx = ip[3];
+                    const int32_t rank = ip[4];
+
+                    MethodTable* arrayClass = (MethodTable*)pMethod->pDataItems[clsIdx];
+
+                    // TODO: Stack slot is 8 bytes aligned and
+                    // AllocateArrayEx expects the dimensions to be in int32_t array
+                    // so we need to convert the stack slots to int32_t
+                    int32_t* dims = (int32_t*)alloca(rank * sizeof(int32_t));
```

Comment:
alloca will overflow the stack when called in the loop. Is it what the TODO above is about? I think this should just be moved to a helper method, so that the alloca buffer gets de-allocated once the opcode executes.